chasm/src/main/java/org/quiltmc/chasm/internal/TransformationSorter.java

```diff
@@ -177,7 +177,7 @@ private static void recurseTargets(List<TargetInfo> targets, int depth, Set<Targ
                 List<TargetInfo> endTargets = targetsByEnd.getOrDefault(index, Collections.emptyList());
 
                 // Remove from enclosing targets
-                enclosingTargets.removeAll(endTargets);
+                endTargets.forEach(enclosingTargets::remove);
```

Comment:
according to intellij, it may be slow
see also: https://stackoverflow.com/questions/28671903/the-hashsett-removeall-method-is-surprisingly-slow#28672583

Reply:
TIL. Maybe you could add a comment referencing that issue in the code ?